app/iot/service.py

```
@@ -1,49 +1,113 @@
 
+    async def register_devices(self, devices: Iterable[BaseDevice]) -> list[str]:
+        await asyncio.gather(*(d.connect() for d in devices))
+        ids: list[str] = []
+        for d in devices:
+            device_id = d.name
+            self._devices[device_id] = d
+            ids.append(device_id)
+        return ids
 
-# Protocol is very similar to ABC, but uses duck typing
-# so devices should not inherit for it (if it walks like a duck, and quacks like a duck, it's a duck)
-class Device(Protocol):
-    def connect(self) -> None:
-        ...  # Ellipsis - similar to "pass", but sometimes has different meaning
+    # ------------------ ОНОВЛЕНО: гнучкий парсер Message ------------------
+    async def send_message(self, message: Any) -> None:
+        """
+        Підтримує різні назви полів у Message:
+          - device_id / device / id
+          - type / message_type / msg_type
+          - payload / data / value
+        Приймає type як Enum (MessageType) або як рядок ("SWITCH_ON"/"flush"/тощо).
+        """
+        device_id = (
+            getattr(message, "device_id", None)
+            or getattr(message, "device", None)
+            or getattr(message, "id", None)
+        )
+        msg_type = (
+            getattr(message, "type", None)
+            or getattr(message, "message_type", None)
+            or getattr(message, "msg_type", None)
+        )
+        payload = (
+            getattr(message, "payload", None)
+            or getattr(message, "data", None)
+            or getattr(message, "value", None)
+        )
 
-    def disconnect(self) -> None:
-        ...
+        if device_id is None:
+            raise AttributeError("Message missing device_id/device/id")
+        if msg_type is None:
+            raise AttributeError("Message missing type/message_type/msg_type")
 
-    def send_message(self, message_type: MessageType, data: str) -> None:
-        ...
+        device = self._devices[device_id]
```

This line obtains the device with `device = self._devices[device_id]`. If `device_id` is not present in `self._devices` this will raise a `KeyError`. It's better to surface a clearer error for callers (or handle it), for example:

- raise a descriptive `KeyError` or `ValueError` with a message like `f"Device {device_id!r} is not registered"`, or
- check with `if device_id not in self._devices:` and raise a controlled exception.

Adding this check will make failures explicit and easier to debug and meets the requirement to "Maintain overall program logic correctness" by not allowing silent crashes in orchestration code that relies on registered devices.